### ISSUE TITLE
Is Masked LARk a (V)DAF?

### ISSUE THREAD

**Issue by cjpatton** [cfrg/draft-irtf-cfrg-vdaf#10]
[This paper](https://arxiv.org/pdf/2110.14794.pdf) describes a distributed computation of a gradient descent. We're given a set of `(x[i], y[i])` pairs, where `x[i]` is client `i`'s *feature* and `y[i]` is the corresponding *label*. From skimming the paper, my understanding is that stochastic gradient descent is computed iteratively as
```
\theta_{j+1} = \theta_j - \eta * (f(x[1], y[1], \theta_j) + ... + f(x[n], y[n], \theta_j))
```
until we find a (locally) optimal model `\theta_j`.  for a particular function `f`. (`\eta` is a constant called the "learning rate".) What the paper shows is that each iteration can be computed by two aggregators who hold the plaintext features `x[1], ..., x[n]`, but only secret shares of the labels `y[1], ..., y[n]`. The features are kept privacy by adding DP.

This looks an awful lot like a distributed aggregation function:
* measurement is a feature and label
* aggregation parameter is the model `\theta_j` from the previous step.
* aggregate result is the model `\theta_{j+1}` for the next step.

And yet there are two important differences:
1. The label is secret shared, but the feature is not. We can certainly formalize security in the usual way --- the aggregators learn nothing about the measurements beyond what they can deduce from their view of the protocol. But this definition isn't all that meaningful if the feature leaks a lot of information about the label.
2. The output isn't verifiable. In particular, the current protocol appears to allow a client to submit bogus input shares and garble the output without being detected.

Adding verifiability (i.e., solving (2)) seems quite feasible. I'm not sure how best to address (1). It might be worth reaching out to the papers' authors to see if they considered how they might keep the features secret.

@csharrison I'm curious if you have thoughts here.

**Comment by cjpatton** [comment]
Another reference to drop here: https://eprint.iacr.org/2018/442.pdf

**Comment by cjpatton** [comment]
One answer to this question is that Prio3 can be used to do an L2-norm check, which is something that's needed for federated learning: https://docs.rs/prio/latest/prio/vdaf/prio3/type.Prio3FixedPointBoundedL2VecSum.html

I think the question I was more after is whether this "apply model to measurement" step can be modeled as (and instantiated with) functional secret sharing.
